### ISSUE TITLE
ExcelPackage fails to work when running on a user account with limited privileges

### ISSUE THREAD

**Issue by MasterEvilAce** [EPPlusSoftware/EPPlus#695]
I am running code under a user account with limited permissions. It does not have access to the application's directory as well as some other things. It DOES have access to the directory where I am trying to create an xlsx file.

The first major hurdle is in ExcelPackage's Init function. It throws an exception when the user account cannot access the appsettings.json file (since the user as no permissions to this directory in my case.)

```
var build = new ConfigurationBuilder()
                .SetBasePath(Directory.GetCurrentDirectory())
                .AddJsonFile("appsettings.json", true,false);  

```
I do not need this functionality personally, and wouldn't mind if this code failing just keeps the defaults and doesn't throw an exception... but if someone does perhaps they'd need to customize the "BasePath" of configuration builder.

While doing more checks, this same code for appsettings.json also appears in IsLicenseSet().
In addition there's some more code that errors here:
```
var v = Environment.GetEnvironmentVariable("EPPlusLicenseContext",EnvironmentVariableTarget.User);
if(string.IsNullOrEmpty(v))
{
  v = Environment.GetEnvironmentVariable("EPPlusLicenseContext", EnvironmentVariableTarget.Process);
}
```
Environment.GetEnvironmentVariable seems to be causing an error "Requested Registry access is not allowed." This seems bypassable by setting ExcelPackage.LicenseContext in code instead of via config, however?

I'm not sure where else EPPlus might have issues with running under a limited user account, but these are major roadblocks for me, and prevents me from getting any further running EPPlus in this way.

**Comment by swmal** [comment]
Thanks for reporting this. We will have a look and will get back asap.

**Comment by swmal** [comment]
We have discussed this and will add a new property to the ExcelPackage class called _SuppressInitializationExceptions_. If this property is set to true EPPlus will catch all Exceptions that are thrown when reading configuration values and environment variables. These Exceptions will then be available via a new property on the ExcelPackage class called InitializationErrors so you can read them.

The reason for not making this the default behaviour of EPPlus is that we believe that it can be a just as common scenario that you would want to fix the cause of these Exceptions instead of ignoring them.

So if you are in an environment with limited privileges you will be able to set this property to true and then set the LicenseContext value via code. It will look something like this:

```csharp
ExcelPackage.LicenseContext = LicenseContext.NonCommercial;
ExcelPackage.SuppressInitializationExceptions = true;
using var package = new ExcelPackage();
if(package.InitializationErrors.Count() > 0)
{
   // do something with the enumerable of errors that where logged during initialization or just ignore them.
}
```
This is still work in progress and will eventually result in a PR on our development branch.

**Comment by swmal** [comment]
This is resolved with 6.0.7. See [this wiki article](https://github.com/EPPlusSoftware/EPPlus/wiki/Configuration) and set the `SuppressInitializationExceptions` property to true.
